### PR TITLE
[bugfix] variables not being passed in tower_job_template module

### PR DESCRIPTION
**Pull request by bmduffy**
##### SUMMARY
I expect that when `ask_inventory: true` that I should not have to supply an `inventory`. Similarly when I set `ask_credential: true` I should be prompted for a credential in the tower UI when I launch a job.
```
- name: "Add job template to tower"
  tower_job_template:
    name: "{{ item.name }}"
    description: "{{ item.description }}"
    job_type: "{{ item.job_type | default('run') }}"
    project: "{{ item.project }}"
    playbook: "{{ item.playbook }}"
    state: present
    ask_limit: true
    ask_inventory: true
    ask_extra_vars: true
    ask_credential: true
  with_items: "{{ tower_jobs }}"
```
Following error occurs when no `inventory` is specified and `ask_inventory: ture`;
```
failed: [host] (item={u'description': u'Test job template', u'project': u'local-project', u'playbook': u'playbooks/example.yaml', u'machine_credential': u'SSH_KEY', u'name': u'Test'}) => {"changed": false, "failed": true, "item": {"description": "Test job template", "machine_credential": "SSH_KEY", "name": "Test", "playbook": "playbooks/example.yaml", "project": "local-project"}, "module_stderr": "Shared connection to host.example.com closed.\r\n", "module_stdout": "Traceback (most recent call last):\r\n  File \"/tmp/ansible_uxpKv0/ansible_module_tower_job_template.py\", line 321, in <module>\r\n    main()\r\n  File \"/tmp/ansible_uxpKv0/ansible_module_tower_job_template.py\", line 308, in main\r\n    result = jt.modify(**params)\r\n  File \"/usr/lib/python2.7/site-packages/tower_cli/models/base.py\", line 961, in modify\r\n    force_on_exists=True, **kwargs)\r\n  File \"/usr/lib/python2.7/site-packages/tower_cli/resources/job_template.py\", line 110, in write\r\n    return super(Resource, self).write(*args, **kwargs)\r\n  File \"/usr/lib/python2.7/site-packages/tower_cli/models/base.py\", line 977, in write\r\n    ret = super(SurveyResource, self).write(pk=pk, **kwargs)\r\n  File \"/usr/lib/python2.7/site-packages/tower_cli/models/base.py\", line 345, in write\r\n    r = getattr(client, method.lower())(url, data=kwargs)\r\n  File \"/usr/lib/python2.7/site-packages/requests/sessions.py\", line 555, in post\r\n    return self.request('POST', url, data=data, json=json, **kwargs)\r\n  File \"/usr/lib/python2.7/site-packages/tower_cli/api.py\", line 235, in request\r\n    kwargs.get('data', None), r.content.decode('utf8'))\r\ntower_cli.exceptions.BadRequest: The Tower server claims it was sent a bad request.\r\n\r\nPOST https://localhost/api/v1/job_templates/\r\nParams: None\r\nData: {\"ask_tags_on_launch\": false, \"job_type\": \"run\", \"playbook\": \"playbooks/techpreview/config.yaml\", \"ask_job_type_on_launch\": false, \"state\": \"present\", \"become_enabled\": false, \"credential\": 3, \"description\": \"Apply techpreview task(s)\", \"ask_variables_on_launch\": true, \"ask_limit_on_launch\": true, \"ask_inventory\": true, \"ask_credential\": false, \"name\": \"[TECHPREVIEW] Apply techpreview tasks\", \"project\": 6, \"tower_verify_ssl\": true}\r\n\r\nResponse: {\"inventory\":[\"Must either set a default value or ask to prompt on launch.\"]}\r\n", "msg": "MODULE FAILURE", "rc": 0}
```
Also updated missing documentation and typo.

##### ISSUE TYPE
 - Bugfix Pull Request

##### COMPONENT NAME
./lib/ansible/modules/web_infrastructure/ansible_tower/tower_job_template.py

##### ANSIBLE VERSION
```
ansible 2.4.0.0
  config file = /home/dapadmin/brian/ansible-tower-management/ansible.cfg
  configured module search path = [u'/home/dapadmin/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python2.7/site-packages/ansible
  executable location = /bin/ansible
  python version = 2.7.5 (default, May  3 2017, 07:55:04) [GCC 4.8.5 20150623 (Red Hat 4.8.5-14)]
```

##### ADDITIONAL INFORMATION
Tested on `ansible 2.4.0.0` but you can see the issue in `ansible:devel` branch.

### CODE REVIEW

**Comment by bmduffy** [comment]
+label ready_for_review 
+label bot_status

**Comment by bmduffy** [comment]
+label ready_for_review

**Review by wwitzel3 — APPROVED** [review]

**Comment by matburt** [comment]
shipit

File: lib/ansible/modules/web_infrastructure/ansible_tower/tower_job_template.py [hunk 1]
```diff
@@ -176,6 +182,8 @@ def update_fields(p):
         'ask_limit': 'ask_limit_on_launch',
         'ask_tags': 'ask_tags_on_launch',
         'ask_job_type': 'ask_job_type_on_launch',
+        'ask_inventory': 'ask_inventory_on_launch',
+        'ask_credential': 'ask_credential_on_launch',
```
Comment (AlanCoding) [inline lib/ansible/modules/web_infrastructure/ansible_tower/tower_job_template.py]: This becomes more complicated with the latest AWX version, but the field_map will have to be refactored in general.

**Comment by AlanCoding** [comment]
shipit

File: lib/ansible/modules/web_infrastructure/ansible_tower/tower_job_template.py [hunk 2]
```diff
@@ -106,6 +106,12 @@
         - Prompt user for extra_vars on launch.
       required: False
       default: False
+    ask_limit:
+      description:
+        - Prompt user for host limits on launch.
+      required: False
+      default: False
+      version_added: "2.5"
```
Comment (ryanpetrello) [inline lib/ansible/modules/web_infrastructure/ansible_tower/tower_job_template.py]: Should this be 2.6 at this point?

**Comment by ryanpetrello** [comment]
Thanks for contributing this, @bmduffy.  I've recently added support for integration tests for the various Tower modules:

https://github.com/ansible/ansible/pull/37421

Would you mind adding similar tests for this bug fix so we can prove it works (and to avoid regressions)?  Thanks!

needs_revision

File: lib/ansible/modules/web_infrastructure/ansible_tower/tower_job_template.py (hunk 2 above)
Comment (gundalow) [reply lib/ansible/modules/web_infrastructure/ansible_tower/tower_job_template.py]: 2.8

**Comment by bmduffy** [comment]
Probably don't need this anymore

**Pull request closed without merging**